src/api.c

```diff
@@ -2714,6 +2714,46 @@ static int cgroup_get_parent_name(struct cgroup *cgroup, char **parent)
 	return ret;
 }
 
+/*
+ * Checks if the cgroup's controller shares the mount point with any other
+ * controller of the cgroup. Returns 1 if shared or 0.
+ */
+static int is_cgrp_ctrl_mnt_shared(struct cgroup *cgroup, char *controller)
+{
+	char *controller_mnt_point = NULL;
+	int ret = 0;
+	int i, j;
+
+	pthread_rwlock_rdlock(&cg_mount_table_lock);
+
+	for (i = 0; cg_mount_table[i].name[0] != '\0'; i++) {
+		if (strncmp(cg_mount_table[i].name, controller,
+			    sizeof(cg_mount_table[i].name)) == 0) {
+
+			controller_mnt_point =
+				strdup(cg_mount_table[i].mount.path);
```

Comment:
Do you need to `strdup` here?  What if you just grab the address of the mount path?
```
controller_mnt_point = &cg_mount_table[i].mount.path;
```

Reply:
Agree, that should be enough.